realm/src/androidTest/java/io/realm/RealmObjectTest.java

```diff
@@ -477,4 +477,35 @@ public void testThreadModelClass() {
         Thread thread = testRealm.createObject(Thread.class);
         testRealm.commitTransaction();
     }
+
+    public void testIsValidUnManagedObject() {
+        AllTypes allTypes = new AllTypes();
+        assertFalse(allTypes.isValid());
+    }
+
+    public void testIsValidClosedRealm() {
+        testRealm.beginTransaction();
+        AllTypes allTypes = testRealm.createObject(AllTypes.class);
+        assertTrue(allTypes.isValid());
+        testRealm.commitTransaction();
+        testRealm.close();
+        assertFalse(allTypes.isValid());
+    }
+
+    public void testIsValidDeletedObject() {
+        testRealm.beginTransaction();
+        AllTypes allTypes = testRealm.createObject(AllTypes.class);
+        assertTrue(allTypes.isValid());
+        testRealm.clear(AllTypes.class);
+        testRealm.commitTransaction();
+        assertFalse(allTypes.isValid());
+    }
+
+    public void testIsValidManagedObject() {
```

Comment:
A little too many almost identical tests imho. This can easily be merged with testIsValidClosedRealm(). But up to you.